Modules/BenchPress.Azure/Tests/Public/Get-ResourceByType.Tests.ps1

```diff
@@ -1,34 +1,56 @@
-﻿BeforeAll {
-  . $PSScriptRoot/../../Public/Confirm-AppServicePlan.ps1
-  . $PSScriptRoot/../../Public/Confirm-ResourceGroup.ps1
-  . $PSScriptRoot/../../Public/Confirm-SqlDatabase.ps1
-  . $PSScriptRoot/../../Public/Confirm-SqlServer.ps1
-  . $PSScriptRoot/../../Public/Confirm-VirtualMachine.ps1
-  . $PSScriptRoot/../../Public/Confirm-WebApp.ps1
+﻿using module ./../../Classes/ResourceType.psm1
+
+BeforeAll {
   . $PSScriptRoot/../../Public/Get-ResourceByType.ps1
+  $testCases = @()
+
+  foreach ($i in [ResourceType].GetEnumNames())
+  {
+    $functionName = "Confirm-$i"
+    $fileName = "$functionName.ps1"
+
+    # dot-source the powershell function
+    . $PSScriptRoot/../../Public/$fileName
+
+    # build test case array
+    $testObject = @{
+      ResourceType = $resource
+      Expected = $functionName
+    }
+
+    $testCases+=$testObject
+  }
 }
 
 Describe "Get-ResourceByType" {
   Context "unit tests" -Tag "Unit" {
     BeforeEach {
```

Comment:
This would probably be better in a `Describe.BeforeAll` because it probably only needs to be run once, and not for each test.

Reply:
good catch!